android/src/main/java/com/reactnativecommunity/asyncstorage/AsyncStorageMigration.java

```diff
@@ -0,0 +1,161 @@
+package com.reactnativecommunity.asyncstorage;
+
+import android.content.Context;
+import android.os.Build;
+import android.util.Log;
+
+import androidx.annotation.RequiresApi;
+
+import com.facebook.react.modules.storage.ReactDatabaseSupplier;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.nio.channels.FileChannel;
+import java.nio.file.Files;
+import java.nio.file.attribute.BasicFileAttributes;
+import java.util.ArrayList;
+
+// A utility class that migrates a scoped AsyncStorage database to RKStorage.
+// This utility only runs if the RKStorage file has not been created yet.
+public class AsyncStorageMigration {
+    static final String LOG_TAG = "ScopedStorageMigration";
```

Comment:
Let's have a tag that's related to AsyncStorage, for example `AsyncStorageExpoMigration`